### DISCUSSION TITLE
Word Splitting functionality

### DISCUSSION THREAD

**Discussion by sunjam** [comictagger/comictagger#299] [Q&A]
Related to #176 Humble names and #297 installing 1.4.4alpha7

I'm testing Word Splitting functionality on alpha7.  So far I have not had a single comic succeed, which seems strange.

1. Open folder containing all .cbz comics from Humblebundle.
2. Right click any comic.  `Lone Wolf and Cub Volume 1- The Assassin's Road - Dark Horse Comics.cbz`

What should the proper naming scheme be for this comic so it can be identified?

Auto-identify.  Auto-tag. Rename.  All of these fail.
Complicated parser?

![image](https://user-images.githubusercontent.com/1787238/169551959-f72242fc-b7c0-4a5b-8be9-557344489197.png)
![image](https://user-images.githubusercontent.com/1787238/169552240-06b48fd1-8f24-472f-9aad-3fa1475ae8e0.png)

A lot of file names include publisher, so I tried adding some to search settings.

![image](https://user-images.githubusercontent.com/1787238/169552621-a45039f0-cd8d-4891-9e62-ee4e9bb61651.png)

Any suggestions for properly renaming `Lone Wolf and Cub Volume 1- The Assassin's Road - Dark Horse Comics.cbz` appreciated.
Or, perhaps I am missing some part of setting up ComicTagger since all tagging attempts fail that I've tried.  :thought_balloon:

**Comment by sunjam** [comment]
[Here is a ~450 entry list of names from Humble Bundle, none of which have been able to work with ComicTagger or Mylar.](https://bin.disroot.org/?64fb4c78b4c0c16c#EDf6vhqV9ZYxKV9eU39eepx2reRTFgG8hVsNjbmVaeWU)

**Reply by sunjam** [reply]
Suggestion: optionally remove non-standard characters from the beginning of filenames.  Example:
`-The Dark Nothing,- in Uptight #5 - Fantagraphics.cbz`
to:
`The Dark Nothing,- in Uptight #5 - Fantagraphics.cbz`

Would it make sense to also replace all hyphens?

or re-write `#5` as `v5`

**Reply by lordwelch** [reply]
So I downloaded my copy of "Lone Wolf and Cub Volume 1: The Assassin's Road" and ended up with the filename `lonewolfandcubvolume1_theassassinsroad.cbz` so I'm not sure how you are getting your filenames but they are better than the ones they come with mostly.

In either case ComicTagger essentially goes in this order:
1. Load comic
1. Find metadata
    1. Look for ComicRack or ComicBookLover tags in the comic
    1. If none are found try to parse the filename
1. Manipulate metadata (there is no specified order beyond this point)
   - Search ComicVine for updated/more complete metadata (auto-tag, auto-identify)
   - Modify metadata manually (the form)
   - Rename files using metadata that was previously obtained

Right now you are still on the try to parse the filename phase my suggestion is to turn on the "Complicated parser" in filename settings, this will parse out more of the filenames correctly. For many of these you may have to add an issue number to get them to search properly you can either type one in manually or when Auto-Tagging there is an option to default the issue to `1`.

For some of them such as `-The Dark Nothing,- in Uptight #5 - Fantagraphics.cbz` you will have to go search around as your not looking for the "correct" title necessarily you are looking for what ComicVine has at labeled under this looks to be https://comicvine.gamespot.com/uptight-5/4000-518513/ which in your naming scheme would be `Uptight #5 - Fantagraphics.cbz`

**Comment by sunjam** [comment]
I could not get Humble re-naming to work via ComicTagger so I ended up renaming the entire library by hand.  Here are my notes... volume naming varied between
```
v1
Vol. 1
_vol3
Volume 1
Vol. #2
issues_issue8
#1
```
Sometimes a hyphen trailed afterwards.

My thoughts for improved scripting, if possible, are...

Re-name all variations above to `#1`
Remove any `.` `,` or `-` at the beginning of a file name.
Remove any `.` `,` within file name.
Remove all text appended after `#1`

So, `-The Dark Nothing,- in Uptight #5 - Fantagraphics.cbz`

becomes

`The Dark Nothing - in Uptight #5.cbz`

or

`The Dark Nothing #5.cbz`

If possible, this could be done before attempting word splitting functionality.  Hope this is helpful.  :smile_cat:

**Reply by lordwelch** [reply]
Glad you were able to tag your comics.
I have made some some changes to the volume handling so in future (pre-)releases it should be handled better.
The "Complicated" parser will parse most of these filenames correctly. You can enable it in [settings]

[settings]: https://github.com/comictagger/comictagger/wiki/Enable-the-%22Complicated%22-parser

The word splitting was already added to ComicTagger and isn't related to your particular issues.
The word splitting was implemented as an optional feature to parse names that are ran together e.g. `hondocitylaw.cbz` which should be `hondo city law.cbz`. Humble Bundle creates these ran together names when you download directly from the website, you appear to be using the [Humble Bundle Downloader] FireFox extension which uses it's own scheme for naming the downloaded files

[Humble Bundle Downloader]: https://addons.mozilla.org/en-US/firefox/addon/humble-bundle-downloader
